python/dataset.cpp

```diff
@@ -147,7 +139,8 @@ void init_dataset(py::module &m) {
   datasetProxy.def(py::init<Dataset &>());
 
   py::class_<Dataset> dataset(m, "Dataset");
-  dataset.def(py::init<>())
```

Comment:
I didn't really understand, doesn't this remove the default constructor `d = Dataset()`?
But then I tried it by checking out the branch and building and `d = Dataset()` still works.
Maybe it is then just using the constructor below (https://github.com/scipp/scipp/blob/7d696fa9a9aa4b7f464c2c10b8fd0029fea6a37b/python/dataset.cpp#L144) where all the arguments have default values?

Reply:
Good catch, I actually removed this by accident! I think you are right though, it will use the one with default arguments instead. So I'd suggest we leave it as it is, unless we run into a reason not to.